### DISCUSSION TITLE
Default values for draw options

### DISCUSSION THREAD

**Discussion by NPatch** [cnr-isti-vclab/meshlab#981] [Ideas] [answered]
Would it be possible to add some configurable defaults?
For example, we tend to use flat shading to view photogrammetry models because it's fast and way lighter than normal shading.
We also want to be able to set culling. We do this all the time and it would be nice to be able to set the defaults.
That said, other properties could also share in this configurable domain.

**Comment by alemuntoni** [comment, marked as answer]
Hi, this is a very nice idea, and makes a lot of sense to have such feature in meshlab.
However the codebase of meshlab concerning shading options needs a huge reorganization to take into account this... Right now I cannot say when this feature will be available..

**Reply by NPatch** [reply]
I see. I'm ok if it makes it into the roadmap. Also one more point where this also applies is, if you change shading options, but you edit the model (e.g. if memory serves, when you select faces and delete), the shading reverts back to the default. Not sure if a bug, but if not, it's also one more point that will need to comply with the configuration of defaults.

**Reply by alemuntoni** [reply]
Yes, I don't know if that behaviour was intended, but it is definitely unexpected and it should be fixed (see #967). There are a lot of similar/related bugs in meshlab :) We will try to do this whenever it will be possible!

**Reply by NPatch** [reply]
Ok. Have a good one.
